### PR TITLE
New data source: materialize_user

### LINKED ISSUE #591: Add `materialize_users` data source to retrive all existing users or filter by user email and get 1 specific user

### PR DESCRIPTION
**Pull request by bobbyiliev**
Adding a new `materialize_user` data source to allow users to retrieve information about existing users by email.

For example at the moment there is no clean way of getting the ID of an existing Frontegg user, this data source will allow exactly this.

Example:

```hcl
data "materialize_user" "example_user" {
  email = "some_email@some_domain.com"
}
output "user" {
  value = data.materialize_user.example_user
}

# Output:
user = {
  "auth_provider" = "google"
  "email" = "some_email@some_domain.com"
  "id" = "unique-frongegg-user-id"
  "verified" = true
}
```

Fixes #591 #590

### CODE REVIEW

File: docs/data-sources/user.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "materialize_user Data Source - terraform-provider-materialize"
+subcategory: ""
+description: |-
+  The user data source allows you to retrieve information about a user in your Frontegg organization.
```
Comment (morsapaes) [inline docs/data-sources/user.md:6]:
```suggestion
  The user data source allows you to retrieve information about a user in your Materialize organization.
```

File: docs/data-sources/user.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+# materialize_user (Data Source)
+
+The user data source allows you to retrieve information about a user in your Frontegg organization.
```
Comment (morsapaes) [inline docs/data-sources/user.md:11]:
```suggestion
The user data source allows you to retrieve information about a user in your Materialize organization.
```

File: docs/data-sources/user.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## Schema
+
+### Required
+
+- `email` (String) The email address of the user to retrieve.
+
+### Read-Only
+
+- `auth_provider` (String) The authentication provider for the user.
+- `id` (String) The unique identifier of the user.
```
Comment (morsapaes) [inline docs/data-sources/user.md:31]:
```suggestion
- `id` (String) The unique (UUID) identifier of the user.
```

File: docs/resources/user.md [hunk 4]
```diff
@@ -3,12 +3,12 @@
 page_title: "materialize_user Resource - terraform-provider-materialize"
 subcategory: ""
 description: |-
-  
+  The user resource allows you to invite and delete users in your Frontegg organization.
```
Comment (morsapaes) [inline docs/resources/user.md:6]:
```suggestion
  The user resource allows you to invite and delete users in your Materialize organization.
```

File: docs/resources/user.md [hunk 5]
```diff
@@ -3,12 +3,12 @@
...
 ---
 
 # materialize_user (Resource)
 
-
+The user resource allows you to invite and delete users in your Frontegg organization.
```
Comment (morsapaes) [inline docs/resources/user.md:11]:
```suggestion
The user resource allows you to invite and delete users in your Materialize organization.
```

File: pkg/datasources/datasource_user.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func User() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: userDataSourceRead,
+
+		Description: `The user data source allows you to retrieve information about a user in your Frontegg organization.`,
```
Comment (morsapaes) [inline pkg/datasources/datasource_user.go:16]:
```suggestion
		Description: `The user data source allows you to retrieve information about a user in your Materialize organization.`,
```

File: pkg/datasources/datasource_user.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+		Schema: map[string]*schema.Schema{
+			"email": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The email address of the user to retrieve.",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique identifier of the user.",
```
Comment (morsapaes) [inline pkg/datasources/datasource_user.go:27]:
```suggestion
				Description: "The unique (UUID) identifier of the user.",
```

File: docs/data-sources/user.md (hunk 1 above)
Comment (morsapaes) [reply docs/data-sources/user.md:6]: FrontEgg is an implementation detail; users shouldn't need to know that this is the authentication provider we use.

**Comment by morsapaes** [comment]
LGTM, but would appreciate a stamp from @benesch here.

File: pkg/datasources/datasource_user.go [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+			"verified": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether the user's email address has been verified.",
+			},
+			"metadata": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Additional metadata associated with the user.",
+			},
```
Comment (benesch) [inline pkg/datasources/datasource_user.go:38]: This field is used internally by Materialize Cloud. It's not strictly speaking sensitive, but it's also not something end users should be looking at. I'd just remove it, if that works!

File: pkg/datasources/datasource_user.go [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+			"auth_provider": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The authentication provider for the user.",
+			},
```
Comment (benesch) [inline pkg/datasources/datasource_user.go:43]: I think this one's likely to be confusing, too. IIUC it's, like, the _last_ auth provider the user used to sign in, so it jumps around a lot.

File: pkg/datasources/datasource_user.go (hunk 8 above)
Comment (bobbyiliev) [reply pkg/datasources/datasource_user.go]: Good points! Just updated the PR to remove both the `metadata` and the `auth_provider` attributes.

**Review by morsapaes — APPROVED** [review]
Thank you, @bobbyiliev!

**Pull request merged**